docs/source/fundamentals/Introduction_to_writer.md

```diff
@@ -0,0 +1,66 @@
+
+1. An `out` parameter is an output dataset directory to save shard files. If the parameter is a local directory path, the shard files are stored locally. If the parameter is a remote directory, a local temporary directory is created to cache the shard files, and then the shard files are uploaded to a remote location. In the end, the temp directory is deleted once shards are uploaded. If the parameter is a tuple of `(local_dir, remote_dir)`, shard files are saved in the `local_dir` and uploaded to a remote location. As shard files are ready, it gets uploaded in the background to a remote location if provided. The user does not have to worry about uploading the shard files manually. `MDSWriter` also support a `keep_local` parameter where after uploading of an individual shard file is completed, you have the flexibility of deleting the shard file locally by providing `keep_local` to `False` (Default is `False`) to avoid running out of disk space.Checkout the [out](https://docs.mosaicml.com/projects/streaming/en/stable/api_reference/generated/streaming.MDSWriter.html) parameter for more detail. For example, one can provide the `out` parameter as shown below:
+<!--pytest.mark.skip-->
+```python
+out = '/tmp/data'
+out = 's3://bucket/data'
+out = {'/local/data', 'oci://bucket/data'}
+```
+
+2. A `column` parameter that maps a feature name or label name with a streaming supported encoding type. `MDSWriter` encodes your data from provided encoding type to bytes, and later it gets decoded back automatically to its original data type when calling `StreamingDataset`. The `index.json` file saves `column` information for decoding. Below is the list of supported encoding formats.
+
+| Name   | Class  | Name    | Class   | Name | Class  |
+| ------ | ------ | ------- | ------- | ---- | ------ |
+| bytes  | `Bytes`  | int8    | `Int8`    | pil  | `PIL`    |
+| str    | `Str`    | int16   | `Int16`   | jpeg | `JPEG`   |
+| int    | `Int`    | int32   | `Int32`   | png  | `PNG`    |
+| uint8  | `UInt8`  | int64   | `Int64`   | pkl  | `Pickle` |
+| uint16 | `UInt16` | float16 | `Float16` | json | `JSON`   |
+| uint32 | `UInt32` | float32 | `Float32` |      |        |
+| uint64 | `UInt64` | float64 | `Float64` |      |        |
+
+Below is one example where the feature name `x` is an image, and the label `y` is a class value.
+<!--pytest.mark.skip-->
+```python
+column = {
+    'x': 'jpeg',
+    'y': 'int8'
+}
+```
+
+**Advance use-case:** If the data type you are interested in is not listed in the above table, then you can write your own data type class with `encode` and `decode` method in it and patch it inside streaming. For example, you want to write the same for `int32` data type.
```

Comment:
Advanced

Reply:
Fixed in https://github.com/mosaicml/streaming/pull/232/commits/9e77cf4f83e0785e964e0df06f847418aa00364a.